Documenting bank failure criteria for us banks from 1934 to present day

https://www.kaggle.com/fdic/bank-failures

What type of banking institution is the most likely to fail? How have bank failure rates changed over time? What commercial bank failure cost the federal government the most to resolve?

In [1]:
#importing libraries

#Organization and Liner algebra
import pandas as pd
import numpy as np

#Visualizations
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline 




#setting filter warnings to ignore to keep our notebook clean
import warnings
warnings.filterwarnings("ignore")


In [2]:
#loading data
data = pd.read_csv('banks.csv')

#data overview
print(data.shape)
data.info()

(3484, 12)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3484 entries, 0 to 3483
Data columns (total 12 columns):
Financial Institution Number    2883 non-null float64
Institution Name                3484 non-null object
Institution Type                3484 non-null object
Charter Type                    3484 non-null object
Headquarters                    3484 non-null object
Failure Date                    3484 non-null object
Insurance Fund                  3484 non-null object
Certificate Number              2999 non-null float64
Transaction Type                3484 non-null object
Total Deposits                  3484 non-null int64
Total Assets                    3333 non-null float64
Estimated Loss (2015)           2509 non-null float64
dtypes: float64(4), int64(1), object(7)
memory usage: 326.8+ KB


Notes: total assets should prob be int but is flt
decide on what cols we need and was arent usefull.....might be able to ignore most of the missing data by not needing columns with missing info

3,484 institutions, 12 columns of info

In [3]:
#looking at numerical data
data.describe()

Financial Institution Number  Certificate Number  Total Deposits  \
count                   2883.000000         2999.000000    3.484000e+03   
mean                    5571.146028        24013.859286    2.997762e+05   
std                     2846.666068        12234.422998    3.304676e+06   
min                     1122.000000           40.000000    0.000000e+00   
25%                     2820.500000        17413.000000    1.229300e+04   
50%                     4592.000000        24846.000000    4.172100e+04   
75%                     7311.500000        30895.500000    1.562415e+05   
max                    11919.000000        91189.000000    1.882608e+08   

       Total Assets  Estimated Loss (2015)  
count  3.333000e+03           2.509000e+03  
mean   4.045884e+05           7.064435e+04  
std    5.454873e+06           3.305872e+05  
min    1.400000e+01          -4.386000e+03  
25%    1.505100e+04           3.852000e+03  
50%    4.752300e+04           1.335400e+04  
75%    1.734780e+05           4.458500e+04  
max    3.070216e+08           1.243800e+07

Really lo

In [5]:
data['Institution Name'].unique()

array(['BANK OF AMERICA TRUST CO.', 'FON DU LAC STATE BANK',
       'FIRST NATIONAL BANK OF LIMA', ..., 'ALLIED BANK',
       'HARVEST COMMUNITY BANK', 'SEAWAY BANK AND TRUST COMPANY'],
      dtype=object)

In [6]:
df = data.copy()
df.columns

Index(['Financial Institution Number', 'Institution Name', 'Institution Type',
       'Charter Type', 'Headquarters', 'Failure Date', 'Insurance Fund',
       'Certificate Number', 'Transaction Type', 'Total Deposits',
       'Total Assets', 'Estimated Loss (2015)'],
      dtype='object')

In [7]:
df = df.drop(columns =['Financial Institution Number','Certificate Number'])

In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3484 entries, 0 to 3483
Data columns (total 10 columns):
Institution Name         3484 non-null object
Institution Type         3484 non-null object
Charter Type             3484 non-null object
Headquarters             3484 non-null object
Failure Date             3484 non-null object
Insurance Fund           3484 non-null object
Transaction Type         3484 non-null object
Total Deposits           3484 non-null int64
Total Assets             3333 non-null float64
Estimated Loss (2015)    2509 non-null float64
dtypes: float64(2), int64(1), object(7)
memory usage: 272.3+ KB


Time to investigate the failure date and see what that looks like.  First we'll make it a date time object, then we'll look at the distribution that shows count of fails by **date**

In [9]:
df['Failure Date'].describe()

count           3484
unique          1639
top       3/16/1989 
freq              51
Name: Failure Date, dtype: object

In [10]:
pd.to_datetime(df['Failure Date'])

0      1934-04-19
1      1934-05-28
2      1934-07-18
3      1934-07-18
4      1934-08-06
          ...    
3479   2016-05-06
3480   2016-08-19
3481   2016-09-23
3482   2017-01-13
3483   2017-01-27
Name: Failure Date, Length: 3484, dtype: datetime64[ns]

In [24]:
df['Failure Month Year'] = pd.to_datetime(df['Failure Date']).dt.to_period('M')


In [30]:
x = df['Failure Month Year'].unique()
y = [df['Failure Month Year'][df['Failure Month Year']==i].count() for i in x]
print(len(x), len(y))

512 512


In [32]:
plt.figure(figsize =(12,15))
plt.plot(x,y)
plt.title('Falures by Date')
plt.show()

TypeError: Axis must have `freq` set to convert to Periods

ValueError: view limit minimum 0.0 is less than 1 and is an invalid Matplotlib date value. This often happens if you pass a non-datetime value to an axis that has datetime units

In [33]:
x

<PeriodArray>
['1934-04', '1934-05', '1934-07', '1934-08', '1934-09', '1934-10', '1934-11',
 '1935-01', '1935-03', '1935-04',
 ...
 '2015-02', '2015-05', '2015-07', '2015-10', '2016-03', '2016-04', '2016-05',
 '2016-08', '2016-09', '2017-01']
Length: 512, dtype: period[M]